Add new information to sidechains page (#6508)
ethereum/ethereum-org-website: src/content/developers/docs/scaling/sidechains/index.md

sidebarDepth: 3
---

> A sidechain is a separate blockchain which runs in parallel to Ethereum; it is connected to Mainnet by a two-way bridge. Sidechains have separate [consensus algorithms](/developers/docs/consensus-mechanisms/), which are often designed for faster processing times. Using a sidechain involves trade-offs, though, as they do not inherit Ethereum's security properties, unlike [layer 2 scaling solutions](/layer-2/). 

Reviewer: We could potentially callout  the scalability trilemma here, and a link to the [vision page](https://ethereum.org/en/upgrades/vision/). As alluded to, these chains often are tuned for scalability, while sacrificing on security or decentralization.

## How do sidechains work? {#how-do-sidechains-work}

> ### Consensus algorithms {#consensus-algorithms}

Reviewer: Before this h3, I would into the "How do sidechains work?" section with a loose overview that these are basically separate blockchains, with their own histories, and their own parameters that make them unique. 

We could then lead into how consensus algorithms are one of the ways they can be unique.

Would also suggest adding information about some other common parameters that are changed, like block gas limits and slot/block times.

Reviewer: More on that last point, we could potentially use this to expand and have a section that mentions how these parameter adjustments impact the decentralization of the network. In short, things like fast block times and high gas limits will give fast/cheap transactions, but will make it significantly harder to operate a node, which can easily make a network/chain with a dangerously small cohort of validators. 

We mention how validators can more easily collude on these chains, this could be the precursor to why that happens.

Author: I added a section to explain block parameters. LMK your thoughts on it.

Reviewer: Looks good! Thanks!

> As mentioned, sidechains use alternative consensus algorithms that are different from Ethereum's proof-of-work (PoW) or proof-of-stake (PoS). Some examples of consensus algorithms used on sidechains include:

Reviewer: There is no exclusion of PoW or PoS for sidechains.

> As mentioned,

I assume you're referring to the intro where you mention:

> Sidechains have separate consensus algorithms, 

I would argue "separate consensus algorithms" is correct, but this does not mean they necessarily fall into a different category of consensus algorithm entirely. 

Would phrase this more than sidechains are in no way connected to Ethereum's consensus, and thus can choose their own. They can use PoS (or even PoW if they really wanted to), or PoA/DPoS, etc

- [Proof-of-authority](https://wikipedia.org/wiki/Proof_of_authority)
- [Delegated proof-of-stake](https://en.bitcoinwiki.org/wiki/DPoS)
- [Byzantine fault tolerance](https://decrypt.co/resources/byzantine-fault-tolerance-what-is-it-explained). 

> Sidechain nodes (called validators) are responsible for ordering and processing transactions, storing the blockchain state, and securing the network. In some cases, validators may be responsible for approving users' entries and exits between the sidechain and Ethereum Mainnet. 

Reviewer: Would argue not all nodes are necessarily validators. Would consider rearranging:
```suggestion
Validating nodes of sidechains are responsible for ordering and processing transactions, storing the blockchain state, and securing the network. In some cases, validators may be responsible for approving users' entries and exits between the sidechain and Ethereum Mainnet. 
```
Also, that last sentence, wouldn't this just be when the bridge contracts are interacted with? Would that be different in some way than a validator just including a transaction in a block? Or are there cases where the validators have to give permission?

Author: Thanks for pointing it out: it looks like I mixed up sidechain validators and bridge validators there. I'll take out the other part.

### EVM compatibility {#evm-compatibility}

> What makes a sidechain particularly exciting is that the chain works the same as the main Ethereum chain because it's based on [the EVM](/developers/docs/evm/). It doesn't merely use Ethereum, it _is_ Ethereum. 

Reviewer: I'm a bit cautious of saying an emphasizing "it _is_ Ethereum." Feel like this could lead to thinking that because a chain is EVM compatible that it "is Ethereum" and adding to the confusion that it's all the same security assurances when it isn't. 

Perhaps we focus on how the language used to interpret smart contracts is the same, allowing for compatibility.

This means if you want to use your [dapp](/developers/docs/dapps/) on a sidechain, it's just a matter of deploying your [smart contract](/developers/docs/smart-contracts/) to this sidechain. It looks, feels, and acts just like Mainnet – you write contracts in Solidity, and interact with the chain via the Web3 API.

> Because sidechains are EVM-compatible, they are considered a useful scaling [scaling solution](/developers/docs/scaling/) for Ethereum-native dapps. With your dapp on a sidechain, users can enjoy lower gas fees and faster transactions (albeit, with certain trade-offs), especially if Mainnet is congested. 

Reviewer: I tend to favor adding a bit of a qualifier to these statements similar to what I mentioned at the beginning. When these chains optimize for scaling, they sacrifice in other areas, and as you can probably tell me now, I personally find that important information to remind people of (since it seems to be easily forgotten 🙂 )

Author: Definitely agree with you on this. I feel this was one of the reasons that made me rework the page: the original didn't really highlight the tradeoffs with using sidechains.

sidebarDepth: 3
---

> A sidechain is a separate blockchain which runs in parallel to Ethereum and is connected to Mainnet by a two-way bridge. Sidechains have separate block parameters and [consensus algorithms](/developers/docs/consensus-mechanisms/), which are often designed for efficient processing of transactions. Using a sidechain involves trade-offs, though, as they do not inherit Ethereum's security properties, unlike [layer 2 scaling solutions](/layer-2/). 

Reviewer 2: ```suggestion
A sidechain is a separate blockchain that runs independent of Ethereum and is connected to Ethereum Mainnet by a two-way bridge. Sidechains can have separate block parameters and [consensus algorithms](/developers/docs/consensus-mechanisms/), which are often designed for efficient processing of transactions. Using a sidechain involves trade-offs, though, as they do not inherit Ethereum's security properties. Unlike [layer 2 scaling solutions](/layer-2/), sidechains do not post state changes and transaction data back to Ethereum Mainnet.
```

Changed to run independently to give a better mental model that these are in fact not connected to Ethereum in any way other than a bridge. I felt like saying they run in parallel gives the impression that they share more data that a bridge. Added a bit more information to the last sentance about what it means to inherit Ethereum Mainnet security.

Sidechains also sacrifice some measure of decentralization or security to achieve high throughput ([scalability trilemma](https://vitalik.ca/general/2021/05/23/scaling.html)). Ethereum is, however, committed to scaling without compromising on decentralization and security as outlined in its [vision statement](/upgrades/vision/) for upgrades.

## How do sidechains work? {#how-do-sidechains-work}

Sidechains are independent blockchains, with different histories, development roadmaps, and design considerations. While a sidechain may share some surface-level similarities with Ethereum, it has several distinctive features.  

### Consensus algorithms {#consensus-algorithms}

> One of the qualities that make sidechains unique (i.e., different from Ethereum) is the consensus algorithm used. Ssidechains don't rely on Ethereum for consensus and can choose alternative consensus protocols that suit their needs. Some examples of consensus algorithms used on sidechains include:

Reviewer 2: ```suggestion
One of the qualities that make sidechains unique (i.e., different from Ethereum) is the consensus algorithm used. Sidechains don't rely on Ethereum for consensus and can choose alternative consensus protocols that suit their needs. Some examples of consensus algorithms used on sidechains include:
```

Small spelling fix 😅

- [Proof-of-authority](https://wikipedia.org/wiki/Proof_of_authority)
- [Delegated proof-of-stake](https://en.bitcoinwiki.org/wiki/DPoS)
- [Byzantine fault tolerance](https://decrypt.co/resources/byzantine-fault-tolerance-what-is-it-explained). 

Like Ethereum, sidechains have validating nodes that verify and process transactions, produce blocks, and store the blockchain state. Validators are also responsible for maintaining consensus across the network and securing it against malicious attacks.

#### Block parameters 

> Ethereum places limits on [block times](/developers/docs/blocks/#block-time) (i.e., the time it takes to produce new blocks) and [block sizes](/developers/docs/blocks/#block-size) (i.e., the number of transactions in each block). Conversely, sidechains often adopt different parameters, such as faster block times and higher gas limits, to achieve high throughput, fast transactions, and low fees. 

Reviewer 2: ```suggestion
Ethereum places limits on [block times](/developers/docs/blocks/#block-time) (i.e., the time it takes to produce new blocks) and [block sizes](/developers/docs/blocks/#block-size) (i.e., the amount of data contained per block denominated in gas). Conversely, sidechains often adopt different parameters, such as faster block times and higher gas limits, to achieve high throughput, fast transactions, and low fees. 
```

While this has some benefits, it has critical implications for network decentralization and security. Block parameters, like fast block times and big block sizes, increase the difficulty of running a full node—leaving a few "supernodes" responsible for securing the chain. In such a scenario, the possibility of validator collusion or a malicious takeover of the chain increases.  

For blockchains to scale without harming decentralization, running a node must be open to everyone—not necessarily parties with specialized hardware. This is why efforts are underway to ensure everyone can [run a full node](/developers/docs/nodes-and-clients/#why-should-i-run-an-ethereum-node) on the Ethereum network. 

### EVM compatibility {#evm-compatibility}

> What makes sidechains useful is their ability to execute contracts developed for the [Ethereum Virtual Machine (EVM)](/developers/docs/evm/). Most EVM-compatible sidechains support smart contracts [written in Solidity](/developers/docs/smart-contracts/languages/), allowing for compatibility with Ethereum Mainnet. 

Reviewer 2: ```suggestion
Some sidechains are EVM-compatible and are able to execute contracts developed for the [Ethereum Virtual Machine (EVM)](/developers/docs/evm/). EVM-compatible sidechains support smart contracts [written in Solidity](/developers/docs/smart-contracts/languages/), as well as other EVM smart contract languages, which means smart contracts written for Ethereum Mainnet will also work on EVM-compatible sidechains. 
```
Just a big of rephrasing around how contracts built for Ethereum will work on EVM compatible sidechains, and not the other way around. But overall like this.

> This means if you want to use your [dapp](/developers/docs/dapps/) on a sidechain, it's just a matter of deploying your [smart contract](/developers/docs/smart-contracts/) to this sidechain. It looks, feels, and acts just like Mainnet – you write contracts in Solidity, and interact with the chain via the Web3 API.

Reviewer 2: ```suggestion
This means if you want to use your [dapp](/developers/docs/dapps/) on a sidechain, it's just a matter of deploying your [smart contract](/developers/docs/smart-contracts/) to this sidechain. It looks, feels, and acts just like Mainnet – you write contracts in Solidity, and interact with the chain via the sidechains RPC.
```

> Because sidechains are EVM-compatible, they are considered a useful scaling [scaling solution](/developers/docs/scaling/) for Ethereum-native dapps. With your dapp on a sidechain, users can enjoy lower gas fees and faster transactions, especially if Mainnet is congested. 

Reviewer 2: ```suggestion
Because sidechains are EVM-compatible, they are considered a useful [scaling solution](/developers/docs/scaling/) for Ethereum-native dapps. With your dapp on a sidechain, users can enjoy lower gas fees and faster transactions, especially if Mainnet is congested. 
```
extra scaling :)

However, as explained previously, using a sidechain involves significant trade-offs. Each sidechain is responsible for its security and doesn't inherit Ethereum's security properties. This increases the possibility of malicious behavior which can affect your users or put their funds at risk. 

### Asset movement {#asset-movement}

> A crucial feature of sidechains is their ability to facilitate seamless transfer of value from and to Ethereum Mainnet. This interoperability with Ethereum is achieved using a blockchain bridge, which enforces a “two-way peg” between two chains. [Bridges](/bridges/) use a network of smart contracts deployed on both networks to control transfer of funds between Ethereum and sidechains. 

Reviewer 2: ```suggestion
In order for a separate blockchain to become a sidechain to Ethereum Mainnet it needs the ability to facilitate the transfer of assets from and to Ethereum Mainnet. This interoperability with Ethereum is achieved using a blockchain bridge. [Bridges](/bridges/) use smart contracts deployed on Ethereum Mainnet and a sidechain to control the bridging of funds between them. 
```

> While bridges help users move funds between Ethereum and the sidechain, the assets are not physically moved across the two chains. Instead, a "lock-mint-burn" mechanism is used for transferring value across chains. Here is a description of the actual process:

Reviewer 2: Ill do some thinking on this section, but there are a few different types of bridges:

1 - Lock and mint
2 - Burn and mint
3 - Atomic swaps (liquidity providing)

Currently there is a PR up to add some bridging content to the developer docs (https://github.com/ethereum/ethereum-org-website/pull/6291, preview: https://ethereumorgwebsitedev01-developerbridgespage.gtsb.io/en/developers/docs/bridges/)
I don't think we need to go through the actual bridging process on this page and can instead link out to here when it goes live.

Reviewer: That PR has been merged in now. 

Thoughts..
1. I think I agree that we should remove that information here, and just leave it in one place on the bridges docs page (now merged in). We can trim most of this out, just mention it uses bridges as a means of migrating assets to and from sidechains, using a variety of mechanisms that you can learn more about in the [bridges documentation]

2. I mentioned this on the other PR, but is there an "burn and unlock" mechanism as alluded to in the below content? Or is the only mechnism used involved burning and "minting" only? If a "burn and unlock" mechanism is used, I would just suggest we add that to the bridge docs in a separate PR.

```suggestion
While bridges help users move funds between Ethereum and the sidechain, the assets are not physically moved across the two chains. Instead, mechanisms that typically involve minting and burning are used for transferring value across chains. More on [how bridges work](/developers/docs/bridges/#how-do-bridges-work).
```

> 
> - **Moving funds from Ethereum to a sidechain**
> 
> Bob (an Ethereum user) wishes to use a sidechain. This is what he does to move funds from Mainnet to the sidechain:
>  
> 1. Bob sends ERC-20 tokens to a smart contract on the Ethereum chain (e.g. Ethereum) and pays the transaction fee. 
> 2. Bob’s ERC-20 tokens are locked up in the smart contract, with the event being relayed to the smart contract on the sidechain.
> 3. Once it receives proof of Bob’s deposit, the sidechain’s smart contract triggers the creation or “minting” of an amount of tokens equal to Bob’s initial deposit. 
> 4. Bob receives the new tokens in his wallet address and can use it to execute transactions on the sidechain. 
> 
> - **Moving funds from a sidechain to Ethereum Mainnet**
> 
> Having concluded his business on the sidechain, Bob wants to withdraw his remaining funds to Ethereum Mainnet. This is what happens:
> 
> 1. Bob sends his remaining tokens to the sidechain’s smart contract for “burning”. Burning is a mechanism for destroying tokens by making them irrecoverable. 
> 2. Information concerning Bob’s deposit transaction and the token burning is relayed to the smart contract on Ethereum. 
> 3. The smart contract then triggers the release of Bob’s ERC-20 tokens on Ethereum, which are sent back to his wallet.
> 
> Note: Bridging funds between Mainnet and sidechains carries its own set of problems (e.g., smart contracts can be hacked), which is another drawback associated with sidechains. More on the [risks of blockchain bridges](/bridges/#bridge-risk). 

Reviewer: ```suggestion
```
